x/ccv/provider/keeper/validator_set_update.go

```diff
@@ -174,11 +128,53 @@ func DiffValidators(
 	return updates
```

Comment:
It seems to me that this `updates` slice is created by iterating on `map`s which are not deterministic.
Shouldn't we sort them, as we do in these [tests](https://github.com/cosmos/interchain-security/blob/330c085024997377d6d01e2c3159f9562a0d913a/x/ccv/provider/keeper/validator_set_update_test.go#L183-L191)?

Reply:
The `updates` are created by executing two `for` loops:
1. `for _, currentVal := range currentValidators {`
2. `for _, nextVal := range nextValidators {`
where `currentValidators` and `nextValidators` are slices. The used maps are used internally in the `for` loops. So I do not believe we would have a non-determinism problem here.

Reply:
You're right. Misread it!